src/custom_nodes/face_blur/face_blur.cpp

```diff
@@ -0,0 +1,259 @@
+    NODE_ASSERT(detectionTensor->dims[0] == 1, "input detection dim[0] must be 1");
+    NODE_ASSERT(detectionTensor->dims[1] == 1, "input detection dim[1] must be 1");
+    NODE_ASSERT(detectionTensor->dims[2] == 200, "input detection dim[2] must be 200");
+    NODE_ASSERT(detectionTensor->dims[3] == 7, "input detection dim[3] must be 7");
+
+    uint64_t _imageHeight = imageTensor->dims[originalImageLayout == "NCHW" ? 2 : 1];
+    uint64_t _imageWidth = imageTensor->dims[originalImageLayout == "NCHW" ? 3 : 2];
+    NODE_ASSERT(_imageHeight <= std::numeric_limits<int>::max(), "image height is too large");
+    NODE_ASSERT(_imageWidth <= std::numeric_limits<int>::max(), "image width is too large");
+    int imageHeight = static_cast<int>(_imageHeight);
+    int imageWidth = static_cast<int>(_imageWidth);
+
+    // Processing
+    if (debugMode) {
+        std::cout << "Processing input tensor image resolution: " << cv::Size(imageHeight, imageWidth) << "; expected resolution: " << cv::Size(originalImageHeight, originalImageWidth) << std::endl;
+    }
+
+    NODE_ASSERT(imageHeight == originalImageHeight, "original image size parameter differs from original image tensor size");
+    NODE_ASSERT(imageWidth == originalImageWidth, "original image size parameter differs from original image tensor size");
+
+    cv::Mat image;
+    if (originalImageLayout == "NHWC") {
+        image = nhwc_to_mat(imageTensor);
+    } else {
+        image = nchw_to_mat(imageTensor);
+    }
+
+    NODE_ASSERT(image.cols == imageWidth, "Mat generation failed");
+    NODE_ASSERT(image.rows == imageHeight, "Mat generation failed");
+
+    uint64_t detectionsCount = detectionTensor->dims[2];
+    uint64_t featuresCount = detectionTensor->dims[3];
+
+    std::vector<cv::Rect> boxes;
+
+    for (uint64_t i = 0; i < detectionsCount; i++) {
+        float* detection = (float*)(detectionTensor->data + (i * featuresCount * sizeof(float)));
+        int imageId = static_cast<int>(detection[0]);
+        int labelId = static_cast<int>(detection[1]);
+        float confidence = detection[2];
+        int xMin = static_cast<int>(detection[3] * imageWidth);
+        int yMin = static_cast<int>(detection[4] * imageHeight);
+        int xMax = static_cast<int>(detection[5] * imageWidth);
+        int yMax = static_cast<int>(detection[6] * imageHeight);
+        if (imageId == 0 && confidence >= confidenceThreshold) {
+            auto box = cv::Rect(cv::Point(xMin, yMin), cv::Point(xMax, yMax));
+            boxes.emplace_back(box);
+            if (debugMode) {
+                std::cout << "Detection:\nImageID: " << imageId << "; LabelID:" << labelId << "; Confidence:" << confidence << "; Box:" << box << std::endl;
+            }
+        }
+    }
+
+    // Applying blur on detected areas
+    for (const auto& box : boxes) {
+        cv::GaussianBlur(image(box), image(box), cv::Size(51, 51), 0);
```

Comment:
What are these 51 s?

Reply:
`cv::Size(51,51)` parameter is kernel size. The bigger the kernel size more pixels from surrounding area are included in median computation. That said blur effect with `cv::Size(5,5)` would be weaker than with `cv::Size(51,51)`. Each pixel with gaussian blur is replaced with the median from surrounding pixels determined by kernel size.

Reply:
Sounds like a parameter we could expose to users. Why not let users set this value in config?
What do you think?